tools/build_arrow.sh

```diff
@@ -78,8 +78,6 @@ function compile_velox_arrow {
             -DARROW_BUILD_SHARED=ON \
             -DARROW_SUBSTRAIT=ON \
             -DARROW_S3=ON \
```

Comment:
Do we still need ARROW_COMPUTE, ARROW_CSV, JSON, SUBSTRAIT, WITH BROTLI, RE2, UTF8PROC , etc?

Parquet write should still need dataset, HDFS, ZIP, etc. right?

Reply:
In velox backend, we can remove the unused dependency.
In Arrow backend, the test is not enough to cover all the cases, and we have much relied on Arrow.
So I will just remove for velox backend, do you agree?

Reply:
Do we use the same build_arrow.sh for velox and arrow backend?

Reply:
Yes, but different build function for each

Reply:
re2 is essential, otherwise will raise error in the runtime
```
Exception in thread "main" java.lang.UnsatisfiedLinkError: /tmp/spark_columnar_plugin_4732148121303368613/libvelox.so: /tmp/spark_columnar_plugin_4732148121303368613/libvelox.so: undefined symbol: _ZN3re23RE2D1Ev

```

Reply:
re2 is essential, otherwise will raise error in runtime, and re2 only used if ARROW_COMPUTE=ON, so ARROW_COMPUTE is also essential
```
Exception in thread "main" java.lang.UnsatisfiedLinkError: /tmp/spark_columnar_plugin_2856861211393915076/libvelox.so: /tmp/spark_columnar_plugin_2856861211393915076/libvelox.so: undefined symbol: _ZN3re23RE2D1Ev
        at java.lang.ClassLoader$NativeLibrary.load(Native Method)
        at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1934)
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1817)
        at java.lang.Runtime.load0(Runtime.java:810)
        at java.lang.System.load(System.java:1088)
        at io.glutenproject.vectorized.JniLibLoader.loadFromPath0(JniLibLoader.java:88)
        at io.glutenproject.vectorized.JniLibLoader.access$300(JniLibLoader.java:57)
        at io.glutenproject.vectorized.JniLibLoader$JniLoadTransaction.loadWithLink(JniLibLoader.java:308)
        at io.glutenproject.vectorized.JniLibLoader$JniLoadTransaction.lambda$commit$1(JniLibLoader.java:258)
```

Reply:
I see. Let's clean Velox's now.